sklearn/utils/validation.py

```diff
@@ -1378,3 +1378,8 @@ def _check_fit_params(X, fit_params, indices=None):
             )
 
     return fit_params_validated
+
+
+def _is_dataframe(X):
+    # Return True if X is a pandas dataframe (or a Series)
+    return hasattr(X, 'iloc')
```

Comment:
This is how we've done in other places as well, but I vaguely remember some upcoming API changes on pandas side which would affect this. How would you test for a DataFrame @TomAugspurger ?

Reply:
Hmm I'm not sure. There aren't any plans to remove `DataFrame.iloc` or `Series.iloc`.

Reply:
Is there a reason why you don't just test for `type(X) == type(pd.DataFrame()) or type(X) == type(pd.Series())`?

Reply:
that would require having pandas as a dependency and we don't want that

Reply:
We can do a similarly direct test without adding a dependency on pandas... But generally duck typing is recommended in python to allow for new players with compatible APIs?